Gems/ROS2/Code/Source/Communication/QoS.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2/Communication/QoS.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ROS2
+{
+    QoS::QoS(const rclcpp::QoS& qos)
+    {
+        m_reliabilityPolicy = qos.reliability();
+        m_durabilityPolicy = qos.durability();
+        m_depth = qos.depth();
+    }
```

Comment:
Member initializer list is preferred for this kind of initialization in this case: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#member-initialization
```suggestion
    QoS::QoS(const rclcpp::QoS& qos)
        : m_reliabilityPolicy(qos.reliability())
        , m_durabilityPolicy(qos.durability())
        , m_depth(qos.depth())
    {
    }
```